packages/utils/src/getOptionMatchingSimpleDiscriminator.ts

```diff
@@ -0,0 +1,36 @@
+import get from 'lodash/get';
+import { PROPERTIES_KEY } from './constants';
+import { RJSFSchema, StrictRJSFSchema } from './types';
+
+export default function getOptionMatchingSimpleDiscriminator<T = any, S extends StrictRJSFSchema = RJSFSchema>(
+  formData: T | undefined,
+  options: S[],
+  discriminatorField?: string
+): number | undefined {
+  if (formData && discriminatorField) {
+    const value = get(formData, discriminatorField);
+
+    if (value === undefined) {
+      return;
+    }
+
+    for (let i = 0; i < options.length; i++) {
+      const option = options[i];
+      const discriminator = get(option, [PROPERTIES_KEY, discriminatorField], {});
+
+      if (discriminator.type === 'object' || discriminator.type === 'array') {
+        continue;
+      }
+
+      if (discriminator.const === value) {
+        return i;
+      }
+
+      if (discriminator.enum?.includes(value)) {
+        return i;
+      }
```

Comment:
As somebody who is not all too familiar with the full `jsonschema` specifiacation, it's hard for me to clearly understand what cases are left after excluding `discriminator.type === 'object' || discriminator.type === 'array'`.

It would make it much more understandable for me, if the function were conceived as `handles consts and enums of primitive js types` rather than `handles all discriminators except discriminator.type === 'object' || discriminator.type === 'array'`. This ties into the naming of the function, and also its docs description.

Reply:
Also, looking at the code, it seems it would produce exactly the same results if the 
``` ts
if (discriminator.type === 'object' || discriminator.type === 'array') {
   continue;
}
```
were to be omitted (for valid schemas), which to me suggests that the function's role should be further refined.

Reply:
After digging a bit deeper, I now understand that discriminator is just a reference to an arbitrary a part of any valid schema, and not a completely independent mechanism of matching options against data. Now it makes sense to me why you wanted to define and name the function this way :)